apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet/Trace/ContextInstanceManager.cs

```diff
@@ -18,27 +18,19 @@
 namespace Google.Cloud.Diagnostics.AspNet
 {
     /// <summary>
-    /// Manages access to the current <see cref="IManagedTracer"/>.
+    /// Manages storage of instances of types for the current <see cref="HttpContext"/>.
     /// </summary>
-    internal static class TracerManager
+    internal static class ContextInstanceManager
     {
-        // The key to save the tracer under.
-        private const string TraceKey = "Google.Cloud.Diagnostics.AspNet.Trace";
-
         /// <summary>
-        /// Sets the current <see cref="IManagedTracer"/>
+        /// Sets the current value for a type.
         /// </summary>
-        public static void SetCurrentTracer(IManagedTracer tracer)
-        {
-            HttpContext.Current.Items[TraceKey] = tracer;
-        }
+        public static void Set<T>(T obj) => 
+            HttpContext.Current.Items[typeof(T)] = obj;
 
         /// <summary>
-        /// Gets the current <see cref="IManagedTracer"/> or null if non exists.
+        /// Gets the current value for a type.
         /// </summary>
-        public static IManagedTracer GetCurrentTracer()
-        {
-            return (IManagedTracer) HttpContext.Current.Items[TraceKey];
-        }
+        public static T Get<T>() => (T)HttpContext.Current.Items[typeof(T)];
```

Comment:
Ok, so there's a potential for a `NullReferenceException` here then if someone specifies a value type that doesn't exist. Maybe it's not a huge concern though.